wolfcrypt/src/aes.c

```diff
@@ -11299,7 +11299,6 @@ int wc_AesCcmEncrypt_ex(Aes* aes, byte* out, const byte* in, word32 sz,
 
 #endif /* HAVE_AESCCM */
 
-#ifndef WOLFSSL_NO_MALLOC
```

Comment:
Why remove this?
If you don't have XMALLOC then you can't compile this.

Reply:
`XMALLOC()` is always available.  It's just a question of what it does.  E.g. if `defined(NO_WOLFSSL_MEMORY) && defined(WOLFSSL_NO_MALLOC)`, and `! (defined(XMALLOC_USER) || defined(XMALLOC_OVERRIDE))`, then `XMALLOC()` is defined as
```
#define XMALLOC(s, h, t)     ((void)(s), (void)(h), (void)(t), NULL)
```

I.e. a do-nothing always-fail stub.

By the way, `WOLFSSL_NO_MALLOC` essentially means don't-call-`malloc()`, per current deployment in the code.  It certainly doesn't mean don't-call-`XMALLOC()`.

In any case the new test scenarios cover the bases on this (`allcryptonly-no-malloc` and `allcryptonly-no-malloc-no-wolf-memory`).